src/Store/StreamDoctrineDbalStore.php

```diff
@@ -283,21 +292,38 @@ function () use ($messages): void {
                     $this->executeSave($columns, $placeholders, $parameters, $types, $this->connection);
                 }
 
-                foreach ($achievedUntilPlayhead as $stream => $playhead) {
+                $subselect = null;
+
+                foreach ($achievedUntilEventId as $stream => $eventId) {
+                    if ($subselect === null) {
+                        if ($this->connection->getDatabasePlatform() instanceof MySQLPlatform) {
+                            $subselect = sprintf(
+                                'SELECT t.id FROM (SELECT * FROM %s) AS t WHERE t.event_id = :event_id',
```

Comment:
Why is there an additional sub select? `(SELECT * FROM %s)`

Reply:
Because mysql... This is what mysql needs. MariaDB fixed it right after the fork xD

Reply:
See: https://dev.mysql.com/doc/refman/8.4/en/subquery-restrictions.html
(second example for UPDATE query)